transports/webrtc-websys/src/stream/poll_data_channel.rs

```diff
@@ -0,0 +1,262 @@
+    /// oneshot since only one onclose event is sent
+    rx_onclose: channel::oneshot::Receiver<()>,
+
+    read_buf_cap: usize,
+}
+
+impl PollDataChannel {
+    /// Constructs a new `PollDataChannel`.
+    pub(crate) fn new(data_channel: RtcDataChannel) -> Self {
+        /*
+         * On Open
+         */
+        let (mut tx_onopen, rx_onopen) = channel::mpsc::channel(2);
+
+        let onopen_callback = Closure::<dyn FnMut(_)>::new(move |_ev: RtcDataChannelEvent| {
+            log::debug!("Data Channel opened");
+            if let Err(e) = tx_onopen.try_send(()) {
+                log::error!("Error sending onopen event {:?}", e);
+            }
+        });
+
+        data_channel.set_onopen(Some(onopen_callback.as_ref().unchecked_ref()));
+        onopen_callback.forget();
+
+        /*
+         * On Message Stream
+         */
+        let (mut tx_onmessage, rx_onmessage) = channel::mpsc::channel(2); // TODO: How big should this be?
+
+        let onmessage_callback = Closure::<dyn FnMut(_)>::new(move |ev: MessageEvent| {
+            let data = ev.data();
+            // Convert from Js ArrayBuffer to Vec<u8>
+            let data = js_sys::Uint8Array::new(&data).to_vec();
+            if let Err(e) = tx_onmessage.try_send(data) {
+                log::error!("Error sending onmessage event {:?}", e)
+            }
+        });
+
+        data_channel.set_onmessage(Some(onmessage_callback.as_ref().unchecked_ref()));
+        onmessage_callback.forget();
+
+        // On Close
+        let (tx_onclose, rx_onclose) = channel::oneshot::channel();
+
+        let onclose_callback = Closure::once_into_js(move |_ev: RtcDataChannelEvent| {
+            log::trace!("Data Channel closed");
+            // TODO: This is Erroring, likely because the channel is already closed by the time we try to send/receive this?
+            if let Err(e) = tx_onclose.send(()) {
+                log::error!("Error sending onclose event {:?}", e);
+            }
+        });
+
+        data_channel.set_onclose(Some(onclose_callback.as_ref().unchecked_ref()));
+        // Note: `once_into_js` Closure does NOT call `forget()`, see the wasm_bindgen::Closure docs for more info
+
+        /*
+         * Convert `RTCDataChannel: bufferedamountlow event` Low Event Callback to Future
+         */
+        let (mut tx_onbufferedamountlow, rx_onbufferedamountlow) = channel::mpsc::channel(2);
+
+        let onbufferedamountlow_callback =
+            Closure::<dyn FnMut(_)>::new(move |_ev: RtcDataChannelEvent| {
+                if let Err(e) = tx_onbufferedamountlow.try_send(()) {
+                    log::warn!(
+                        "Sending onbufferedamountlow failed, channel is probably closed {:?}",
+                        e
+                    )
+                }
+            });
+
+        data_channel
+            .set_onbufferedamountlow(Some(onbufferedamountlow_callback.as_ref().unchecked_ref()));
+        onbufferedamountlow_callback.forget();
```

Comment:
I can confirm that `bufferedamountlow` event is indeed firing, so I'm not sure if A) the docs are outdated/incorrect for Chrome or if B) since we don't explicitly **set** `bufferedAmountLowThreshold` to anything (including `0`) then this is avoided? 

We could compare to how firefox behaves

Reply:
> We could compare to how firefox behaves

Did you manage to have this work in Firefox at all? The example doesn't for me.